pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -314,6 +319,31 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeRegi
 	return nil
 }
 
+// Comma-separated list of clusters (or * for any) with an istiod that should
+// attempt leader election for a remote cluster.
+const istiodClusterAnnotation = "topology.istio.io/controlPlaneClusters" // TODO make proper API annotation.TopologyControlPlaneClusters.Name
+
+// checkShouldLead returns true if the caller should attempt leader election for a remote cluster.
+func (m *Multicluster) checkShouldLead(client kubelib.Client, systemNamespace string) bool {
+	if features.ExternalIstiod {
+		namespace, err := client.Kube().CoreV1().Namespaces().Get(context.TODO(), systemNamespace, metav1.GetOptions{})
```

Comment:
this means that a transient error will permanently break things, and if a user adds the label 1s late it will also never recover. Seems like we should be using a namespace informer?

Reply:
It will make things a little bit complicated

Reply:
@howardjohn Added a TODO and made it fail open in the error case for now, so this PR is  NOOP in that case. PTAL

Reply:
This doesn't feel like a NOP to me. It seems like very unsafe behavior. Failing open is not safe either, since now we will lead when we are not supposed to

Reply:
@frankbu, any updates on the followup to use a namespace informer?